proposals/NNNN-GetInteriorVehicleData.md

```diff
@@ -0,0 +1,96 @@
+- If parameter `subscribe=true` and the app has already subscribed to the module, SDL cut off `subscribe=true` from the request and forward it to HMI.
+- If parameter `subscribe=false` and the app has not subscribed to the module yet, SDL cut off  `subscribe=false` from the request and forward it to HMI.
+- If parameter `subscribe=false` and the app has already subscribed to the module, SDL forward the whole request to HMI.
+
+### Advantages:
+-	A simple SDL. Just forwarding the GetInteriorVehicleData requests and response between the app and the HMI.
+
+### Shortcomings:
+-	The operation of GetInteriorVehicleData and subscribe InteriorVehicleData on HMI side may be a time consuming task. 
+-	If app1 has already subscribed data change notification for a moduleType, SDL still need to include parameter `subscribe=true` when forwarding requests from other apps, i.e. to re-subscribe the data again. There are extra messages and extra processing time.
+-	If app1 has not subscribed data change notification for a moduleType, SDL still need to include parameter `subscribe=false` when forwarding requests from other apps, i.e. to un-subscribe the data per app. There are extra messages and extra processing time.
+-	If app1 has already subscribed data change notification for a moduleType, and HMI returns the data to SDL in response,  SDL still forwards new GetInteriorVehicleData requests (from same app or different app regarding same moduleType) to HMI. However, SDL has the module data already and can keep the data up to date since SDL will get OnInteriorVehicleData notifications whenever there is any data change in the same moduleType. There are extra messages and extra processing time.
+-	It is not very clean. SDL still maintains the local status and make some filtering (cut off parameter `subscribe` in certain cases), why not always forward original message if we want to keep SDL simple.
+
+
+## Proposed solution
+
+The following sequence diagram shows the proposed behavior of sdl regarding GetInteriorVehicleData. 
+
+![../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_new_subscribe.png](../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_new_subscribe.png)
+
+SDL maintains the module (Type) subscription status as a whole. SDL need to subscribe to a module if there is at least one app subscribes to the module. SDL need to un-subscribe to a module if there is no app subscribes to the module. Or equivalently
+-	SDL shall only include parameter `subscribe=true` if SDL has not (equivalently there is no app) subscribed to the module data when forwarding GetInteriorVehicleData with `subscribe=true`.
+-	SDL shall only include parameter `subscribe=false` if it is the last app to unsubscribe to the module data when forwarding GetInteriorVehicleData with `subscribe=false`.
+
+
+SDL maintains a cache of module data only when there is at least an application subscribed to the module. 
```

Comment:
Please update to "...at least one application..."

Reply:
changed